auto_submit/lib/server/request_handler.dart

```diff
@@ -25,4 +27,79 @@ abstract class RequestHandler {
   Future<Response> run(Request request) async {
     throw const MethodNotAllowed('GET');
   }
+
+  /// Gets the value associated with the specified [key] in the request
+  /// context.
+  ///
+  /// Concrete subclasses should not call this directly. Instead, they should
+  /// access the getters that are tied to specific keys, such as [request]
+  /// and [response].
+  ///
+  /// If this is called outside the context of an HTTP request, this will
+  /// throw a [StateError].
+  @protected
+  U? getValue<U>(RequestKey<U> key, {bool allowNull = false}) {
```

Comment:
We should avoid pulling this code in as it adds a lot of complexity

Reply:
Deleted unused logic in request_handler.